### ISSUE TITLE
copy-tex modifying double-click selection behaviour

### ISSUE THREAD

**Issue by alexjbest** [KaTeX/KaTeX#1981]
> Before reporting a bug
- [x] Check [common issues](https://katex.org/docs/issues.html).
- [x] Check the bug is reproducible in [the demo](https://katex.org). If not, check KaTeX is up-to-date and installed correctly.
- [x] Search for [existing issues](https://github.com/KaTeX/KaTeX/issues).

**Describe the bug:**
Copy-tex modifies copy behaviour so that a hidden div with text content sandwiched between a visible div with text content and a katex element is also selected when the visible div is double clicked. See steps below:

**To Reproduce:**
1. Go to https://codepen.io/alexjbest/pen/XwaQgR
2. Double click Hello! to select the whole line
3. Copy the selection with ctrl-c or cmd-c or whatever
4. Paste somewhere and see that "Hello!goodbye!" was in the clipboard.
5. Go to codepen settings and disable copy-tex.js and try again, step 4. will only copy Hello.

**Expected behavior:**
"Hello!" should be the contents of the clipboard even when copy-tex is enabled.

**Workaround:**
If I add a &nbsp; between the two div's the behaviour is as expected, but I really think copy-tex shouldn't necessitate this.

**Environment (please complete the following information):**
 - KaTeX Version: latest 0.10.2
 - Device: desktop
 - OS: macOS 10.13.6
 - Browser: Chrome
 - Version: 74

I can also reproduce this in safari with the same system.

This may be related to #1973 though the fix there doesn't seem to work for me.

While this is a slightly specific setup it really happens on http://www.lmfdb.xyz/Genus2Curve/Q/169/a/169/1 for instance, when clicking Show commands for Magma and trying to copy-paste the first block of magma code one inadvertently also selects the hidden sage code.

**Comment by edemaine** [comment]
I reproduced this on Chrome on Windows, with triple clicking (the gesture for line selection).  I imagine it's something in `copy-tex.js`, unrelated to #1973; will take a look.

**Comment by alexjbest** [comment]
I just noticed another (probably the same/related issue.

On a page such as http://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/144/1/g/a/ we have  a section for q-expansion, where there are two options, see up to 10 terms or up to 100 terms, switching between these is toggleable with a click of a link which hides one (style="display: none;") and shows the other, with copy-tex, click dragging to select and copying copies both the hidden and visible one.

Again the issue is related to copy-tex copying things in hidden divs, which normally cannot happen.

**Comment by edemaine** [comment]
@alexjbest Yes, that's a better description of the problem. I did investigate this back in May, but then dropped the ball. I believe `copy-tex` needs to be rewritten substantially to take the following approach:
* Modify the DOM/style in-place to hide what we don't want (math HTML) and show what we want (math LaTeX).  Alternatively, it might be possible to copy the DOM elsewhere before making this modifications, but I'm not sure it makes too much difference, as this will all happen before the browser has a chance to display anything.
* Use [`innerText`](http://perfectionkills.com/the-poor-misunderstood-innerText/) to get the text formatting as it would be in regular copy/paste.
* Put the DOM/style back as it was.

`innerText` should take into account whether something is visible or hidden, and generally seems to behave like copy/paste does, whereas the current plugin implements very different behavior (because I didn't originally realize the difference -- sorry!).

If anyone has time to try implementing this, it would be appreciated!  Otherwise I will try to get to it in the future (currently traveling).
